lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    
+    if @services.include?('Vehicle Registration') == true
+      @registered_vehicles << vehicle
+      vehicle.set_registration_date
+      vehicle.set_plate_type
+
+      if vehicle.antique? == true
```

Comment:
When you have a long method like this, think about how you can use helper methods to break up the logic. For example, you could have a helper method here that you call `process_antique_registration` to handle setting the plate type and collecting the right amount of fees for this type of car.

Using helper methods can help a lot when we have complex or nested conditionals, because those get hard to read. Extract some of that logic into a separate location and give it a readable name to indicate to another dev what is happening.